src/core/ChainRequest.js

```diff
@@ -0,0 +1,56 @@
+import fetch from 'node-fetch';
+
+const api_port = '4000';
+const url = 'http://localhost:';
+
+const request = (route, method, port = api_port, body = {}) => {
+  return new Promise(async (resolve, reject) => {
+    let result;
+    if (method === 'GET') {
+      result = await fetch(url + port + route, { method });
+    } else {
+      result = await fetch(url + port + route, { method, body });
+    }
+    const { status, ...data } = await result.json();
+    !status ? resolve(data) : reject(data);
+  });
+};
+
+export const listAllChains = () => {
+  return request(`/chains/`, 'GET');
+};
+
+export const listAllSnapshots = (chainType = 'ganache') => {
+  return request(`/chain/snapshots/${chainType}`, 'GET');
+};
+
+// if this chain is not active, the server will throw an error
+export const getChainInfo = id => {
+  return request(`/chain/${id}`, 'GET');
+};
+
+export const deleteChain = id => {
+  return request(`/chain/${id}`, 'DELETE');
+};
+
+export const downloadSnapshot = id => {
+  return request(`/chain/snapshot/${id}`, 'GET');
+};
+
+export const getBlockNumber = port => {
+  return request(
+    '/',
+    'POST',
+    port,
+    '{"jsonrpc":"2.0","method":"eth_blockNumber","params":[],"id":1}'
```

Comment:
is it ok for `id` to always be 1?

Reply:
https://www.jsonrpc.org/specification
https://ethereum.stackexchange.com/questions/50647/json-rpc-api-id

I don't think it matters too much.

#### id
An identifier established by the Client that MUST contain a String, Number, or NULL value if included. If it is not included it is assumed to be a notification. The value SHOULD normally not be Null [1] and Numbers SHOULD NOT contain fractional parts [2]
The Server MUST reply with the same value in the Response object if included. This member is used to correlate the context between the two objects.

Reply:
ah, i see, it can be used to match responses to requests, but we don't care about that so much here since we're just sending one request at a time. ok then